src/mgmt/srv_drpc.c

```diff
@@ -435,16 +484,8 @@ ds_mgmt_drpc_pool_create(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 		goto out;
 	}
 
-	D_ASSERT(svc->rl_nr > 0);
-
-	rc = rank_list_to_uint32_array(svc, &resp.svc_reps, &resp.n_svc_reps);
-	if (rc != 0)
-		D_GOTO(out_svc, rc);
-
-	D_DEBUG(DB_MGMT, "%d service replicas\n", svc->rl_nr);
+	pool_create_fill_resp(&resp, pool_uuid, svc);
 
-out_svc:
-	d_rank_list_free(svc);
```

Comment:
[Nit] Since the "pool_create_fill_resp" name does not imply that the function consumes the svc parameter, it would be normal to still free svc in ds_mgmt_drpc_pool_create.

Reply:
- [ ] Free svc parameter in `ds_mgmt_drpc_pool_create()` function body.